src/encoded/commands/release_file.py

```diff
@@ -0,0 +1,491 @@
+from __future__ import annotations
+import argparse
+from typing import Any, Dict, List, Union, Tuple
+import pprint
+
+pp = pprint.PrettyPrinter(indent=2)
+
+from dcicutils import ff_utils
+from dcicutils.creds_utils import SMaHTKeyManager
+
+
+##################################################################
+##################################################################
+##
+##  The file release will do the following updates to the metadata
+##  - Set file status to `released`
+##  - Associate the file with the fileset that the corresponding
+##    submitted files are in
+##  - Adds `dataset`` and `access_status`` to the file
+##  - Set the associated QualityMetrics item and the metrics.zip
+##    file to status `released`
+##  - Set corresponding FileSet to `released`
+##  - Set FileSet associated libraries, and sequencing to `released`
+##  - Set library associated assay and analyte to `released`
+##  - Set analyte associated samples to `released`
+##  - Set sample associated sample_source to `released`
+##
+##################################################################
+##################################################################
+
+
+class PC:  # PortalConstants
+    ACCESSION = "accession"
+    ACCESS_STATUS = "access_status"
+    AGE = "age"
+    ALIGNED_READS = "Aligned Reads"
+    ALIGNMENT_DETAILS = "alignment_details"
+    ANALYTE = "analyte"
+    ANNOTATED_FILENAME = "annotated_filename"
+    ASSAY = "assay"
+    CONSORTIA = "consortia"
+    DATA_CATEGORY = "data_category"
+    DATASET = "dataset"
+    EXTRA_FILES = "extra_files"
+    FILE_SETS = "file_sets"
+    FILE_SET = "file_set"
+    FILENAME = "filename"
+    FINAL_OUTPUT = "Final Output"
+    GERMLINE_VARIANT_CALLS = "Germline Variant Calls"
+    IDENTIFIER = "identifier"
+    LIBRARIES = "libraries"
+    LIBRARY = "library"
+    OUTPUT_STATUS = "output_status"
+    OPEN = "Open"
+    PROTECTED = "Protected"
+    QUALITY_METRICS = "quality_metrics"
+    QUALITY_METRIC = "quality_metric"
+    RELEASED = "released"
+    SAMPLE_SOURCES = "sample_sources"
+    SAMPLES = "samples"
+    SAMPLE = "sample"
+    SEQUENCING = "sequencing"
+    SEQUENCING_CENTER = "sequencing_center"
+    SEQUENCING_READS = "Sequencing Reads"
+    SOMATIC_VARIANT_CALLS = "Somatic Variant Calls"
+    STATUS = "status"
+    SUBMISSION_CENTERS = "submission_centers"
+    SUBMITTER_ID = "submitter_id"
+    SUBMITTED_ID = "submitted_id"
+    TYPE = "@type"
+    UPLOADED = "uploaded"
+    UUID = "uuid"
+    VARIANT_TYPE = "variant_type"
+    VERSION = "version"
+
+
+# dataset is required but comes in through input args for now
+REQUIRED_FILE_PROPS = [PC.SEQUENCING_CENTER]
+
+
+class FileRelease:
+
+    def __init__(self, auth_key):
+        self.key = auth_key
+        self.patch_infos = []
+        self.patch_dicts = []
+        self.warnings = []
+        self.file_accession = ""
+
+    def prepare(self, file_identifier: str, dataset: str):
```

Comment:
Use of type hints are a bit all over the place in this module; consider all or none.